xpti/include/xpti/xpti_data_types.h

```diff
@@ -468,6 +494,22 @@ enum class trace_event_type_t : uint16_t {
   offload_buffer = XPTI_EVENT(9),
   /// Indicates that the current event is an offload accessor related
   offload_accessor = XPTI_EVENT(10),
+  /// Indicates that the current event is an offload sampled image related
+  offload_sampled_image = XPTI_EVENT(11),
+  /// Indicates that the current event is an offload unsampled image related
+  offload_unsampled_image = XPTI_EVENT(12),
+  /// Indicates that the current event is an offload sampled image accessor
+  /// related
+  offload_sampled_image_accessor = XPTI_EVENT(13),
+  /// Indicates that the current event is an offload sampled image host accessor
+  /// related
+  offload_host_sampled_image_accessor = XPTI_EVENT(14),
+  /// Indicates that the current event is an offload unsampled image accessor
+  /// related
+  offload_unsampled_image_accessor = XPTI_EVENT(15),
+  /// Indicates that the current event is an offload unsampled image host
+  /// accessor related
+  offload_host_unsampled_image_accessor = XPTI_EVENT(16),
```

Comment:
The Event types are primarily used to indicate if the "busy" time is attributed to the scheduler, a barrier or an algorithm. For memory operations, this would be "algorithm" as it is needed by an algorithm or we can have a new category called "memory_operation".  We don't anything more fine grain than that.